### PR TITLE
Add switch of DocMs step on each artifact

### PR DESCRIPTION
**Pull request by sima-zhu**
Added feature of enable/disable DocMs step per artifact level.
Issue:https://github.com/Azure/azure-sdk-tools/issues/922

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ stages:
           - ${{if ne(artifact.options.skipPublishDocs, 'true')}}:
             - deployment: PublicDocsMs
               displayName: "Publish Updates for Docs.MS"
-              condition: and(succeeded(), ne(variables['Skip.PublishDocs'], 'true'))
+              condition: and(succeeded(), ne(variables['Skip.PublishDocs'], 'true'), ne(artifact.skipDocMs, 'true'))
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:150]: Why can you not do this in the if statement at template time?

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 2]
```diff
@@ -147,7 +147,7 @@ stages:
           - ${{if ne(artifact.options.skipPublishDocs, 'true')}}:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:147]:
```suggestion
          - ${{if ne(artifact.skipPublishDocMs, 'true')}}:
```

File: eng/pipelines/templates/stages/archetype-java-release.yml (hunk 1 above)
Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-java-release.yml:150]: I did not skip it successfully for the first run. But it skips the step after kicking off another build.
I will move to the if clause.

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-java-release.yml:150]: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=532666&view=results

It indicates put it under if clause works well

File: eng/pipelines/templates/stages/archetype-java-release.yml (hunk 2 above)
Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-java-release.yml:147]: Do we want to control this by DocMs only? GithubIO is still using `options.skipPublishDocs`. Then we have two vars for two doc steps.

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-java-release.yml:147]: I'd change the github.io publishing `skipPublishDocGithubIo` and we probably should remove the "options." from all these conditions.

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 3]
```diff
@@ -209,7 +209,7 @@ stages:
                           # we override the regular script path because we have cloned the build tools repo as a separate artifact.
                           ScriptPath: 'eng/common/scripts/copy-docs-to-blobstorage.ps1'
 
-          - ${{if ne(artifact.options.skipUpdatePackageVersion, 'true')}}:
+          - ${{if ne(artifact.skipUpdatePackageVersion, 'true')}}:
             - deployment: UpdatePackageVersion
               displayName: "Update Package Version"
               condition: and(succeeded(), ne(variables['Skip.UpdatePackageVersion'], 'true'))
```
Comment (scbedd) [inline eng/pipelines/templates/stages/archetype-java-release.yml:215]: Same `Integration` skip question as seen in the other languages..

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-java-release.yml:215]: Talked offline. Will add check on integration step here as well.

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 4]
```diff
@@ -48,7 +48,7 @@ stages:
   - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
     - ${{ each artifact in parameters.Artifacts }}:
       - stage: Release_${{artifact.safeName}}
-        displayName: 'Release: ${{artifact.name}}'
+        displayName: 'Release: ${{artifact.name}}__${{artifact.skipPublishDevFeed}}'
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:51]: This looks like a copy/past issue.

File: sdk/template/ci.yml [hunk 5]
```diff
@@ -30,3 +30,4 @@ extends:
       - name: azure-sdk-template
         groupId: com.azure
         safeName: azuresdktemplate
+        skipPublishDevFeed: true
```
Comment (weshaggard) [inline sdk/template/ci.yml:33]: I'm fine with this for testing but we should undo it before merging this PR.

**Review by weshaggard — APPROVED** [review]

**Review by scbedd — APPROVED** [review]

**Pull request merged**